### ISSUE TITLE
Add unpythonic.it.pad

### ISSUE THREAD

**Issue by Technologicat** [Technologicat/unpythonic#23]
Occasionally, padding with a fill value is needed for things such as chunking. Currently, this is what we get:

```python
from unpythonic import chunked

chunks = chunked(3, range(9))
assert [tuple(chunk) for chunk in chunks] == [(0, 1, 2), (3, 4, 5), (6, 7, 8)]
chunks = chunked(3, range(7))
assert [tuple(chunk) for chunk in chunks] == [(0, 1, 2), (3, 4, 5), (6,)]
```

What if we wanted ``[(0, 1, 2), (3, 4, 5), (6, None, None)]``, for example to be able to always ``a, b, c = chunk``?

To keep things orthogonal, it could be useful to provide a separate ``pad(n, fillvalue, iterable)`` function, which pads an iterable with copies of ``fillvalue`` if its length is smaller than ``n``. Then we could write:

```python
from unpythonic import chunked, pad  # "pad" doesn't exist yet

chunks = chunked(3, range(7))
chunks = [pad(3, None, chunk) for chunk in chunks]
assert [tuple(chunk) for chunk in chunks] == [(0, 1, 2), (3, 4, 5), (6, None, None)]
```

Or even:

```python
from unpythonic import chunked, curry, pad  # "pad" doesn't exist yet

chunks = chunked(3, range(7))
pad_to_3_with_None = curry(pad, 3, None)
chunks = [pad_to_3_with_None(chunk) for chunk in chunks]
assert [tuple(chunk) for chunk in chunks] == [(0, 1, 2), (3, 4, 5), (6, None, None)]
```

This can be implemented in terms of ``itertools.islice`` and ``collections.deque``, a bit similarly to ``butlastn`` or ``lastn``.

**Comment by Technologicat** [comment]
Sketching a bit... a simple solution:

```python
def pad(n, fillvalue, iterable):
    k = 0  # used if iterable is empty
    for k, x in enumerate(iterable, start=1):
        yield x
    for _ in range(k, n):
        yield fillvalue

assert tuple(pad(5, None, range(3))) == (0, 1, 2, None, None)
assert tuple(pad(5, None, ())) == (None, None, None, None, None)
```

**Comment by Technologicat** [comment]
Resolved in 16e683b.
